File: app/src/main/java/nl/invissvenska/qrscanner/service/QRTileService.java
package nl.invissvenska.qrscanner.service;

import android.content.Intent;
import android.service.quicksettings.TileService;

public class QRTileService extends TileService {
    @Override
    public void onClick() {
        super.onClick();
        Intent intent = new Intent("nl.invissvenska.qrscanner.MainActivity");
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        startActivityAndCollapse(intent);
    }

    @Override
    public void onTileRemoved() {
        super.onTileRemoved();
    }

    @Override
    public void onTileAdded() {
        super.onTileAdded();
    }

    @Override
    public void onStartListening() {
        super.onStartListening();
    }

    @Override
    public void onStopListening() {
        super.onStopListening();
    }
}

File: app/src/main/java/nl/invissvenska/qrscanner/service/QRTileService.kt
package nl.invissvenska.qrscanner.service

import android.content.Intent
import android.service.quicksettings.TileService

class QRTileService : TileService() {
    override fun onClick() {
        super.onClick()
        val intent = Intent("nl.invissvenska.qrscanner.MainActivity")
        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
        startActivityAndCollapse(intent)
    }

    override fun onTileRemoved() {
        super.onTileRemoved()
    }

    override fun onTileAdded() {
        super.onTileAdded()
    }

    override fun onStartListening() {
        super.onStartListening()
    }

    override fun onStopListening() {
        super.onStopListening()
    }
}